lib/chatbot-api/chatbot-dynamodb-tables/index.ts

```diff
@@ -19,8 +25,14 @@ export class ChatBotDynamoDBTables extends Construct {
         type: dynamodb.AttributeType.STRING,
       },
       billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
-      encryption: dynamodb.TableEncryption.AWS_MANAGED,
-      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      encryption: props.kmsKey
+        ? dynamodb.TableEncryption.CUSTOMER_MANAGED
+        : dynamodb.TableEncryption.AWS_MANAGED,
+      encryptionKey: props.kmsKey,
```

Comment:
What happens if props.kmsKey does not exist?

Reply:
It passes undefined. (property is ignored)